linode/manager
# upcoming: [DI-21119] - Handle jwe token limit in ACLP UI and Bug fix for resources component

## Description 📝

 Handled jwe token limit in ACLP UI and Bug fix for resources component.

## Changes  🔄

List any change(s) relevant to the reviewer.

- JWE token will only be created for selected resources/database clusters. It was earlier created for all the available resources of selected service type.
- Resources/Database clusters autocomplete has a bug fix - 
     When these three are equal => 1. available resources/clusters 2. selected resources/clusters 3. max resource selection limit, select/deselect all will work as expected.  Earlier, in this edge case, deselect all got disabled after maximum resources selection. 


## Target release date 🗓️

2-Dec-2024

## Preview 📷

| Before  | After   |
| ------- | ------- | 
| ![image](https://github.com/user-attachments/assets/53f1db03-5f37-4c69-ad19-1c75e35798df) | ![image](https://github.com/user-attachments/assets/28d2ba87-6f4b-44d5-ae68-5eff6e77199d) |

## How to test 🧪

### Verification steps

(How to verify changes)

- Navigate to monitor tab.
- Select all the filters. 
- Open networks tab, see the token call fired, observe the payload passed, it has selected resources/clusters' ids.


## As an Author, I have considered 🤔

- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


---
